utils.h

```diff
@@ -0,0 +1,9 @@
+#ifndef UTILS_H
+#define UTILS_H
+
+#include "mkmimo.h"
```

Comment:
Is this necessary?  I think it's better to keep the header hierarchy/dependency flatter.

Actually, I see why you had it there.  Maybe it's better to declare `Buffer` type here and call this buffer.h?

Also header for `size_t` should be included here.